cache/class-vary-cache.php

```diff
@@ -344,6 +341,21 @@ private static function set_cookie( $name, $value ) {
 	private static function unset_cookie( $name ) {
 		setcookie( $name, '', time() - 3600 );
 	}
+	/**
+	 * Only allow alphanumerics, dash and underscore
+	 *
+	 * @param string $value The string you want to test on.
+	 * @return boolean
+	 */
+	private static function validate_cookie_values( $value ) {
+		if ( preg_match( '/[^_0-9a-zA-Z-]+/', $value ) > 0 ) {
+			return 'Invalid character(s). Can only use alphanumerics, dash and underscore';
```

Comment:
Should probably return a `WP_Error` object here rather than a string?

Also, tests?

Reply:
You're right, I was thinking it should be up to the calling function in how it wants to deal with the errors. But we can do:
```
if ( is_wp_error( $value) ) {
  trigger_error(...
```
in the calling functions instead.

It's tested indirectly from the tests I added in the register_groups() but I should probably add explicit tests too.

Reply:
Done